src/components/tile/tile.js

```diff
@@ -0,0 +1,96 @@
+import mixin from '../../globals/js/misc/mixin';
+import createComponent from '../../globals/js/mixins/create-component';
+import initComponentBySearch from '../../globals/js/mixins/init-component-by-search';
+import eventMatches from '../../globals/js/misc/event-matches';
+
+class Tile extends mixin(createComponent, initComponentBySearch) {
+  /**
+   * Tile.
+   * @extends CreateComponent
+   * @extends InitComponentBySearch
+   * @param {HTMLElement} element The element working as an Tile.
+   */
+  constructor(element, options) {
+    super(element, options);
+    this.tileType = this.element.dataset.tile;
+    this.tileHeight = 0; // Tracks expandable tile height
+    this.atfHeight = 0; // Tracks above the fold height
+    this._hookActions(this._getClass(this.tileType));
+  }
+
+  _getClass = type => {
+    const typeObj = {
+      expandable: this.options.classExpandedTile,
+      clickable: this.options.classClickableTile,
+      selectable: this.options.classSelectableTile,
+    };
+    return typeObj[type];
+  };
+
+  _hookActions = tileClass => {
+    const isExpandable = this.tileType === 'expandable';
+    if (isExpandable) {
+      const aboveTheFold = this.element.querySelector(
+        this.options.selectorAboveTheFold
+      );
+      if (aboveTheFold) {
+        this.tileHeight = this.element.getBoundingClientRect().height;
+        this.atfHeight = aboveTheFold.getBoundingClientRect().height;
+        this.element.style.maxHeight = `${this.atfHeight}px`;
+      }
+    }
+    this.element.addEventListener('click', evt => {
+      const input = eventMatches(evt, this.options.selectorTileInput);
+      if (!input) {
+        this.element.classList.toggle(tileClass);
+      }
+      if (isExpandable) {
+        this._setTileHeight();
+      }
+    });
+    this.element.addEventListener('keydown', evt => {
```

Comment:
this `keydown` event listener breaks the native keyboard behavior of the `.bx--tile__chevron` button. While this block exists, the button doesn't work with the keyboard for expanding tiles.

Reply:
yep, you actually only need to add `onClick` handlers for `buttons` - onClick actually maps to 'default' action, which for most elements is click. For buttons, default actions map to click, enter, and space - in your tests, `click` tests cover all three of those